library/core/src/ptr/mut_ptr.rs

```diff
@@ -2219,102 +2251,120 @@ impl<T: ?Sized> PartialOrd for *mut T {
 mod verify {
     use crate::kani;
 
-    macro_rules! generate_mut_add_and_sub_harness {
-        ($type:ty, $proof_name:ident, $func_name:ident) => {
-            #[kani::proof_for_contract(<*mut $type>::$func_name)]
+    macro_rules! generate_mut_arithmetic_harness {
+        ($type:ty, $proof_name:ident, add) => {
+            #[kani::proof_for_contract(<*mut $type>::add)]
             pub fn $proof_name() {
                 let mut test_val: $type = kani::any::<$type>();
+                let offset: usize = kani::any();
+                let count: usize = kani::any();                
+                kani::assume(offset <= 1);
+                
                 let test_ptr: *mut $type = &mut test_val;
+                let ptr_with_offset: *mut $type = test_ptr.wrapping_add(offset);                   
+                unsafe {
+                    ptr_with_offset.add(count);
+                }
+            }
+        };
+        ($type:ty, $proof_name:ident, sub) => {
+            #[kani::proof_for_contract(<*mut $type>::sub)]
+            pub fn $proof_name() {
+                let mut test_val: $type = kani::any::<$type>();
+                let offset: usize = kani::any();
                 let count: usize = kani::any();
+                kani::assume(offset <= 1);
+                
+                let test_ptr: *mut $type = &mut test_val;
+                let ptr_with_offset: *mut $type = test_ptr.wrapping_add(offset);
+                unsafe {
+                    ptr_with_offset.sub(count);
+                }
+            }
+        };
+        ($type:ty, $proof_name:ident, offset) => {
+            #[kani::proof_for_contract(<*mut $type>::offset)]
+            pub fn $proof_name() {
+                let mut test_val: $type = kani::any::<$type>();
+                let offset: usize = kani::any();
+                let count: isize = kani::any();
+                kani::assume(offset <= 1);
+
+                let test_ptr: *mut $type = &mut test_val;
+                let ptr_with_offset: *mut $type = test_ptr.wrapping_add(offset);                
```

Comment:
do you think we can use wrapping_byte_add to create and test unaligned pointers?

Reply:
I think it should work, but we need to wait until same_allocation is updated to test it

Reply:
Just tested it out, actually unaligned pointer worked fine, I think we can use `wrapping_byte_add` in our next version